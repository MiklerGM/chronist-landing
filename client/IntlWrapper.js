import React from 'react';
import PropTypes from 'prop-types';
import { IntlProvider } from 'react-intl';

import localeDataRU from './locales/ru.json';
// import localeDataEN from './locales/en.json';

class IntlWrapper extends React.Component {
  render() {
    return (
      <IntlProvider
        locale={this.props.locale}
        key={this.props.locale}
        messages={this.props.messages}
      >
        {this.props.children}
      </IntlProvider>
    );
  }
}

IntlWrapper.propTypes = {
  locale: PropTypes.string,
  messages: PropTypes.object
};

IntlWrapper.defaultProps = {
  locale: 'ru',
  messages: localeDataRU
};

export default IntlWrapper;
