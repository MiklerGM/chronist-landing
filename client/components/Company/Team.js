import React from 'react';
import { FormattedMessage } from 'react-intl';

import './Company.less';

import dataRU from './data/TeamDataRU';
import dataEN from './data/TeamDataEN';

const Person = ({ person }) => (
  <div className='person--data'>
    <img src={person.img} alt={person.name} />
    <p>
      <strong>
        {person.name}
      </strong>
      <br />
      {person.position}
      <br />
      <a href={`mailto://${person.link}`}>{person.link}</a>
    </p>
    <br />
  </div>
);

class Team extends React.Component {
  render() {
    let data = {};
    if (this.props.locale === 'ru') {
      data = dataRU;
    } else {
      data = dataEN;
    }

    return (
      <div id='team' className='page--segment'>
        <div className='page--content'>
          <h2 className='invert'>
            <span><FormattedMessage id='team.title' /></span>
          </h2>
          <div className='person--list'>
            {data.map((person) => <Person person={person} key={`personkey_${person.id}`} />)}
          </div>
        </div>
      </div>
    );
  }
}

export default Team;
