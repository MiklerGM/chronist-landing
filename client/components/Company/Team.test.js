import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import IntlWrapper from '../../IntlWrapper';
import Team from './Team';

const TestComponent = () => (
  <IntlWrapper>
    <Router>
      <Team onChangeLanguage={(e) => console.log('dump func', e)} locale='en' />
    </Router>
  </IntlWrapper>
);

// it must render
it('Team render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

it('Team render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

// Snapshot tests
it('Team Snapshot', () => {
  const tree = renderer
    .create(<TestComponent />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
