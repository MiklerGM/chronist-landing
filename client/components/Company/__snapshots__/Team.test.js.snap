// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team Snapshot 1`] = `
<div
  className="page--segment"
  id="team"
>
  <div
    className="page--content"
  >
    <h2
      className="invert"
    >
      <span>
        Наша команда
      </span>
    </h2>
    <div
      className="person--list"
    >
      <div
        className="person--data"
      >
        <img
          alt="Daniil"
          src="test-file-stub"
        />
        <p>
          <strong>
            Daniil
          </strong>
          <br />
          Data Engineering
          <br />
          <a
            href="mailto://data@chronist.ru"
          >
            data@chronist.ru
          </a>
        </p>
        <br />
      </div>
      <div
        className="person--data"
      >
        <img
          alt="Alisa"
          src="test-file-stub"
        />
        <p>
          <strong>
            Alisa
          </strong>
          <br />
          Product Owner
          <br />
          <a
            href="mailto://idea@chronist.ru"
          >
            idea@chronist.ru
          </a>
        </p>
        <br />
      </div>
      <div
        className="person--data"
      >
        <img
          alt="Michael"
          src="test-file-stub"
        />
        <p>
          <strong>
            Michael
          </strong>
          <br />
          Software Engineering
          <br />
          <a
            href="mailto://code@chronist.ru"
          >
            code@chronist.ru
          </a>
        </p>
        <br />
      </div>
      <div
        className="person--data"
      >
        <img
          alt="Valeria"
          src="test-file-stub"
        />
        <p>
          <strong>
            Valeria
          </strong>
          <br />
          Visual Design
          <br />
          <a
            href="mailto://"
          >
            
          </a>
        </p>
        <br />
      </div>
      <div
        className="person--data"
      >
        <img
          alt="Dmitry"
          src="test-file-stub"
        />
        <p>
          <strong>
            Dmitry
          </strong>
          <br />
          Frontend Engineering
          <br />
          <a
            href="mailto://"
          >
            
          </a>
        </p>
        <br />
      </div>
    </div>
  </div>
</div>
`;
